app/assets/javascripts/copo/copo.edit-checkin.es6

```diff
@@ -1,54 +1,111 @@
     $editable.attr('contenteditable', true);
-    $editable.focus();
-    document.execCommand('selectAll', false, null);
-    COPO.editCheckin.setEditableListeners($editable)
+
+    if ($editable.hasClass("date")) {
+      // if user edits date input set datepicker and open
+      COPO.editCheckin.setDatepicker($editable).pickadate("open");
+    } else {
+      // select all the text to make it easier to edit
+      $editable.focus();
+      document.execCommand('selectAll', false, null);
+
+      // mousing over the map shows crosshair to quickly set latlng
+      $('#map').addClass('crosshair');
+      map.on('click', function(e) {
+        COPO.editCheckin.handleMapClick($editable, e);
+      });
+    }
+
+    // setup other listeners
+    COPO.editCheckin.setEditableListeners($editable);
   },
 
   setEditableListeners($editable) {
     var original = $editable.text();
 
-    // mousing over the map shows crosshair to quickly set latlng
-    $('#map').toggleClass('crosshair');
-    map.on('click', function(e) {
-      COPO.editCheckin.handleMapClick($editable, e);
-    });
-
-    // if they click the popup, stop editing
+    // if user clicks the popup, stop editing
     $('.leaflet-popup').on('click', function (e) {
       if (e.target.className !== 'editable') {
-        COPO.editCheckin.handleCoordsEdited(original, $editable);
+        COPO.editCheckin.handleEdited(original, $editable);
       }
     });
 
-    // if they hit enter or esc, stop editing
+    // if user hits enter or esc, stop editing
     $editable.on('keydown', function (e) {
       if (e.which === 27 || e.which === 13 ) {
-        COPO.editCheckin.handleCoordsEdited(original, $editable);
+        COPO.editCheckin.handleEdited(original, $editable);
       }
     });
 
-    // if they click another marker, remove all the listeners
+    // if user clicks another marker, remove all the listeners
     COPO.maps.allMarkers.eachLayer(function(marker) {
       marker.on('click', function(e) {
-        if ($editable.attr('contenteditable')) {
-          COPO.editCheckin.handleEditEnd($editable);
-        }
+        COPO.editCheckin.handleEditEnd($editable);
       });
     });
   },
 
+  setDatepicker($editable) {
+    return $("body").pickadate({
+      selectMonths: true,
+      selectYears: 15,
+      closeOnSelect: true,
+      onSet: function(context) {
+        if ("select" in context) {
+          if (this.get("value")) {
+            let date = new Date($editable.text());
+            let newDate = new Date(this.get("value"));
+
+            date.setDate(newDate.getDate());
+            date.setMonth(newDate.getMonth());
+            date.setFullYear(newDate.getFullYear());
+
+            // open marker popup back again and set new date
+            $editable.text(
+              date.toDateString() + ' ' + date.toLocaleTimeString() + " UTC+0000"
+            );
+
+            // remove datepicker with respect to next one
+            this.stop();
```

Comment:
destroy current datepicker in the end of the session, so next time it can be reattached to new popup